pkg/reconciler/kubernetes/progress/ready.go

```diff
@@ -120,15 +121,45 @@ func getLatestReplicaSet(ctx context.Context, deployment *appsv1.Deployment, cli
 		return nil, nil
 	}
 
-	sort.Sort(replicaSetsByCreationTimestamp(ownedReplicaSets))
-	return ownedReplicaSets[len(ownedReplicaSets)-1], nil
+	return FindNewReplicaSet(deployment, ownedReplicaSets), nil
 }
 
-type replicaSetsByCreationTimestamp []*appsv1.ReplicaSet
+// FindNewReplicaSet returns the new RS this given deployment targets (the one with the same pod template).
+func FindNewReplicaSet(deployment *appsv1.Deployment, rsList []*appsv1.ReplicaSet) *appsv1.ReplicaSet {
```

Comment:
This one should not be exported either :)

Reply:
fully agree, fixed